### PR TITLE
Refactor cs-base email operations and upgrade javax.mail library

### PR DESCRIPTION
**Pull request by norberth-csorba**
**User Story ID:** 
720742

**Description:**
Added following inputs to all email operations:
- timeout
- proxyHost
- proxyPort
- proxyUsername
- proxyPassword
Moved all operations from _oo-base-legacy-plugin.mail_ to _cs-actions.cs-mail_.
Refactored to remove code duplication and make it more readable.

**Testing done:**
Unit tests.
Automation Testing.

### CODE REVIEW

File: cs-mail/src/main/java/io/cloudslang/content/mail/utils/MessageStoreUtils.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        SSLContext context = SSLContext.getInstance(SecurityConstants.SSL);
+        context.init(keyManagers, trustManagers, new SecureRandom());
+        SSLContext.setDefault(context);
+    }
+
+
+    public static Store tryTLSOtherwiseTrySSL(Authenticator auth, Properties props, GetMailInput input) throws MessagingException {
+        Store store = configureStoreWithTLS(props, auth, input);
+        try {
+            store.connect(input.getHostname(), input.getUsername(), input.getPassword());
```
Comment (lrevnic) [inline cs-mail/src/main/java/io/cloudslang/content/mail/utils/MessageStoreUtils.java]: Recommendation generated by Amazon CodeGuru Reviewer

This code contains a potential information leakage in the error handling for the following call(s): &#39;GetMailInput.getPassword()&#39; (line 152). You are handling this error data through: &#39;Store.connect()&#39; (line 152) [with catch clause(s) at: [line 153]]. Please use your organization-specific sensitive information handling loggers or redaction. [Learn more about information leakage](https://cwe.mitre.org/data/definitions/209.html) (external link).

File: cs-mail/src/main/java/io/cloudslang/content/mail/services/SendMailService.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+            NoSuchProviderException, SMIMEException {
+        if (input.isEncryptedMessage()) {
+            mimeBodyPart = gen.generate(mimeBodyPart, input.getEncryptionAlgorithm(), SecurityConstants.BOUNCY_CASTLE_PROVIDER);
+        }
+        return mimeBodyPart;
+    }
+
+    private void addEncryptionSettings() throws Exception {
+        URL keystoreUrl = new URL(input.getEncryptionKeystore());
+        InputStream publicKeystoreInputStream = keystoreUrl.openStream();
```
Comment (lrevnic) [inline cs-mail/src/main/java/io/cloudslang/content/mail/services/SendMailService.java]: Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *publicKeystoreInputStream*. The resource is referenced in statements at the following line: 251. The resource closure statement is at line: 253. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by KeyStore.getInstance. Either a) close *publicKeystoreInputStream* in a try-finally block or b) close the resource by declaring *publicKeystoreInputStream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

File: cs-mail/src/main/java/io/cloudslang/content/mail/services/GetMailAttachmentService.java [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        try {
+            is = part.getInputStream();
+
+            File f = new File(path);
+            if (!f.exists() || overwrite) {
+                if (!f.createNewFile()) {
+                    throw new IOException("Could not create file at path: " + path);
+                }
+
+                os = new FileOutputStream(f);
```
Comment (lrevnic) [inline cs-mail/src/main/java/io/cloudslang/content/mail/services/GetMailAttachmentService.java]: Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *os*. The resource is referenced in statements at the following lines: 255, 270, 274, 280. The resource closure statement is at line: 281. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by InputStream.close. Either a) close *os* in a try-finally block or b) close the resource by declaring *os* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).

Comment (norberth-csorba) [reply cs-mail/src/main/java/io/cloudslang/content/mail/services/GetMailAttachmentService.java]: false positive

**Pull request merged**